nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/MoveHDFS.java

```diff
@@ -357,9 +357,10 @@ public Object run() {
                         if (destinationExists) {
                             switch (processorConfig.getConflictResolution()) {
                                 case REPLACE_RESOLUTION:
-                                    if (hdfs.delete(file, false)) {
+                                    // Remove destination file (newFile) to replace
```

Comment:
Maybe we could add some unit tests, something like this:
```java
@Test
    public void testPutWhenAlreadyExistingShouldFailWhenFAIL_RESOLUTION() throws IOException {
        testPutWhenAlreadyExisting(MoveHDFS.FAIL_RESOLUTION, MoveHDFS.REL_FAILURE, "randombytes-1");
    }

    @Test
    public void testPutWhenAlreadyExistingShouldIgnoreWhenIGNORE_RESOLUTION() throws IOException {
        testPutWhenAlreadyExisting(MoveHDFS.IGNORE_RESOLUTION, MoveHDFS.REL_SUCCESS, "randombytes-1");
    }

    @Test
    public void testPutWhenAlreadyExistingShouldReplaceWhenREPLACE_RESOLUTION() throws IOException {
        testPutWhenAlreadyExisting(MoveHDFS.REPLACE_RESOLUTION, MoveHDFS.REL_SUCCESS, "randombytes-2");
    }

    private void testPutWhenAlreadyExisting(String conflictResolution, Relationship expectedDestination, String expectedContent) throws IOException {
        // GIVEN
        Files.createDirectories(Path.of(INPUT_DIRECTORY));
        Files.createDirectories(Path.of(OUTPUT_DIRECTORY));
        Files.copy(Path.of(TEST_DATA_DIRECTORY, "randombytes-2"), Path.of(INPUT_DIRECTORY, "randombytes-1"));
        Files.copy(Path.of(TEST_DATA_DIRECTORY, "randombytes-1"), Path.of(OUTPUT_DIRECTORY, "randombytes-1"));

        MoveHDFS processor = new MoveHDFS();

        TestRunner runner = TestRunners.newTestRunner(processor);
        runner.setProperty(MoveHDFS.INPUT_DIRECTORY_OR_FILE, INPUT_DIRECTORY);
        runner.setProperty(MoveHDFS.OUTPUT_DIRECTORY, OUTPUT_DIRECTORY);
        runner.setProperty(MoveHDFS.CONFLICT_RESOLUTION, conflictResolution);

        byte[] expected = Files.readAllBytes(Path.of(TEST_DATA_DIRECTORY, expectedContent));

        // WHEN
        runner.enqueue(new byte[0]);
        runner.run();

        // THEN
        runner.assertAllFlowFilesTransferred(expectedDestination);

        List<MockFlowFile> flowFiles = runner.getFlowFilesForRelationship(expectedDestination);
        Assert.assertEquals(1, flowFiles.size());

        byte[] actual = Files.readAllBytes(Path.of(OUTPUT_DIRECTORY, "randombytes-1"));

        assertArrayEquals(expected, actual);
    }
```

Reply:
Ok @tpalfy . Thank you! I've just put your code in MoveHDFS test class.